receiver/dockerreceiver/config_test.go

```diff
@@ -0,0 +1,117 @@
+			expected: createDefaultConfig(),
+		},
+		{
+			id: component.NewIDWithName(metadata.Type, "allsettings"),
+			expected: &Config{
+				Endpoint:         "http://example.com/",
+				DockerAPIVersion: "1.40",
+				Stats: StatsConfig{
+					ControllerConfig: scraperhelper.ControllerConfig{
+						CollectionInterval: 2 * time.Second,
+						InitialDelay:       time.Second,
+						Timeout:            20 * time.Second,
+					},
+					ExcludedImages: []string{
+						"undesired-container",
+						"another-*-container",
+					},
+
+					ContainerLabelsToMetricLabels: map[string]string{
+						"my.container.label":       "my-metric-label",
+						"my.other.container.label": "my-other-metric-label",
+					},
+
+					EnvVarsToMetricLabels: map[string]string{
+						"MY_ENVIRONMENT_VARIABLE":       "my-metric-label",
+						"MY_OTHER_ENVIRONMENT_VARIABLE": "my-other-metric-label",
+					},
+				},
+
+				MetricsBuilderConfig: func() metadata.MetricsBuilderConfig {
+					m := metadata.DefaultMetricsBuilderConfig()
+					m.Metrics.ContainerCPUUsageSystem = metadata.MetricConfig{
+						Enabled: false,
+					}
+					m.Metrics.ContainerMemoryTotalRss = metadata.MetricConfig{
+						Enabled: true,
+					}
+					return m
+				}(),
+			},
+		},
+	}
+
+	for _, tt := range tests {
+		t.Run(tt.id.String(), func(t *testing.T) {
+			cm, err := confmaptest.LoadConf(filepath.Join("testdata", "config.yaml"))
+			require.NoError(t, err)
+
+			factory := NewFactory()
+			cfg := factory.CreateDefaultConfig()
+
+			sub, err := cm.Sub(tt.id.String())
+			require.NoError(t, err)
+			require.NoError(t, component.UnmarshalConfig(sub, cfg))
+
+			assert.NoError(t, component.ValidateConfig(cfg))
+			if diff := cmp.Diff(tt.expected, cfg, cmpopts.IgnoreUnexported(metadata.MetricConfig{}), cmpopts.IgnoreUnexported(metadata.ResourceAttributeConfig{})); diff != "" {
+				t.Errorf("Config mismatch (-expected +actual):\n%s", diff)
+			}
+		})
+	}
+}
+
+func TestValidateErrors(t *testing.T) {
+	cfg := &Config{Stats: StatsConfig{ControllerConfig: scraperhelper.NewDefaultControllerConfig()}}
+	assert.Equal(t, "endpoint must be specified", component.ValidateConfig(cfg).Error())
+
+	cfg = &Config{
+		DockerAPIVersion: "1.21",
+		Endpoint:         "someEndpoint",
+		Stats: StatsConfig{
+			ControllerConfig: scraperhelper.ControllerConfig{CollectionInterval: 1 * time.Second},
+		},
+	}
+	assert.Equal(t, `"api_version" 1.21 must be at least 1.25`, component.ValidateConfig(cfg).Error())
+
+	cfg = &Config{
+		Endpoint:         "someEndpoint",
+		DockerAPIVersion: "1.25",
+		Stats: StatsConfig{
+			ControllerConfig: scraperhelper.ControllerConfig{},
+		},
+	}
+	// The message is generated twice because when it's nested,
+	// it is returned both as an error of ControllerConfig and StatsConfig.
+	assert.Equal(t, `"collection_interval": requires positive value; "collection_interval": requires positive value`, component.ValidateConfig(cfg).Error())
+}
```

Comment:
Two errors are reported here. I checked with debugger and it seems that it's because `ControllerConfig` is nested inside `StatsConfig`.